frontend/src/components/CardInfo.tsx

```diff
@@ -18,6 +23,46 @@ const CardInfo = ({
   description,
   price,
 }: Props) => {
+    const userId: number = useSelector((state: Rootstate) => state.user.userId);
+    
+    const [isFavorite, setIsFavorite] = useState(false);
+
+    const addToFavorites = async () => {
+        console.log(userId);
+        if (userId == -1 || !userId) {
+          console.error('User is not logged in');
+          return;
+        } 
+
+        try {
+            const authToken = localStorage.getItem('authToken');
+
+            const options: object = {
+              method: 'POST',
+              headers: {
+                  "Content-Type": "application/json",
+                  'Authorization': `Bearer ${authToken}`
+              },
+              body: JSON.stringify({
+                userId,
+                barName: name,
+                address,
+                note: '',
+                imageURL: image,
+              })
+            };
+
+            const response: Response = await fetch(addFav, options);
+
+            if (response.ok) {
+                setIsFavorite(true);
+            } else {
+                console.error('Failed to add to favorites');
```

Comment:
Read the response instead and display the error message. I.e

const res = await response.json();
console.log(res.error);